quickwit-directories/src/bundle_directory.rs

```diff
@@ -51,6 +52,24 @@ fn split_footer(file_slice: FileSlice) -> io::Result<(FileSlice, FileSlice)> {
 }
 
 impl BundleDirectory {
+    /// Get files and their sizes in a split.
+    pub fn get_stats_split(data: Bytes) -> io::Result<Vec<(PathBuf, usize)>> {
```

Comment:
I'm sure there is a good reason but I don't understand why we are not opening a bundle  with a fileslice and then call `get_stats_split` and avoid all this
```
        let split = OwnedBytes::new(data.to_vec());
        let split_file = FileSlice::new(Box::new(split));
        let (body_and_bundle_metadata, hot_cache) = split_footer(split_file)?;
        let file_offsets =
            BundleStorageFileOffsets::open_from_file_slice(body_and_bundle_metadata)?;
```

Reply:
and when I see `data: Bytes` as the argument, I wonder how to obtain those bytes to get the stats. Should I read the file?

Reply:
The bundle directory does not contain the hotcache and we don't have a split bundle representation elsewhere in the code. For that reason I would prefer to put the hotcache into the BundleDirectory, but it would undo a change from Paul, that removed the hotcache.

About the bytes, that's what the storage returns. I wasn't sure how to get a file slice there. The idea was to use the resolver to easily read splits on s3.

Reply:
arf, it's annoying :/. We need a plan on that, it seems unclear.